### PR TITLE
Compatibility with v6. 

### LINKED ISSUE #32: Compatibility with Autofac v6
v6 Compatibility and version dependency checks ahead of release.

### PR DESCRIPTION
**Pull request by alistairjevans**
Also stopped using a brand new container instance every time we load a module just to find constructors!

### CODE REVIEW

File: src/Autofac.Configuration/Autofac.Configuration.csproj [hunk 1]
```diff
@@ -18,10 +19,8 @@
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
     <RepositoryType>git</RepositoryType>
     <RepositoryUrl>https://github.com/autofac/Autofac.Configuration</RepositoryUrl>
-    <GenerateAssemblyTitleAttribute>false</GenerateAssemblyTitleAttribute>
-    <GenerateNeutralResourcesLanguageAttribute>false</GenerateNeutralResourcesLanguageAttribute>
-    <GenerateAssemblyCopyrightAttribute>false</GenerateAssemblyCopyrightAttribute>
-    <GenerateAssemblyDescriptionAttribute>false</GenerateAssemblyDescriptionAttribute>
+    <AssemblyTitle>Autofac.Configuration</AssemblyTitle>
+    <NeutralLanguage>en-GB</NeutralLanguage>
```
Comment (tillig) [inline src/Autofac.Configuration/Autofac.Configuration.csproj]: All the other assemblies default to `en-US`.

Comment (tillig) [reply src/Autofac.Configuration/Autofac.Configuration.csproj]: (Nice try, buddy!) 😆

**Comment by tillig** [comment]
Yeah, I'm not sure what was up with that constructor selector thing. It's changed a lot over time going from manual XML parsing config to using `IConfiguration` and I'm guessing I just moved the code without thinking about what it was doing. Where's that forehead-slap emoji...?

File: test/Autofac.Configuration.Test/Autofac.Configuration.Test.csproj [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">net461;netcoreapp2.0</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netcoreapp2.0</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">net472;netcoreapp3.0</TargetFrameworks>
```
Comment (tillig) [inline test/Autofac.Configuration.Test/Autofac.Configuration.Test.csproj]: I don't think we need to keep the net472 project tests around, do we? I think this is an artifact of DNX and things not working; and we don't have a desktop .NET target in Autofac core anymore. (I mean, we still exclude `*.xproj` down on line 19, right? Love that DNX.)

**Review by tillig — CHANGES_REQUESTED** [review]
A couple minor things, otherwise looking good.

File: src/Autofac.Configuration/Autofac.Configuration.csproj (hunk 1 above)
Comment (alistairjevans) [reply src/Autofac.Configuration/Autofac.Configuration.csproj]: 🙄

**Review by tillig — APPROVED** [review]
:gb:

**Pull request merged**